lib/modules/manager/npm/post-update/yarn.ts

```diff
@@ -56,16 +56,22 @@ export async function checkYarnrc(
       const yarnBinaryExists = yarnPath
         ? await localPathIsFile(yarnPath)
         : false;
+      let scrubbedYarnrc = yarnrc
+        .replace('--install.pure-lockfile true', '')
+        .replace('--install.frozen-lockfile true', '');
       if (!yarnBinaryExists) {
-        const scrubbedYarnrc = yarnrc.replace(
+        scrubbedYarnrc = scrubbedYarnrc.replace(
           regEx(/^yarn-path\s+"?.+?"?$/gm),
           ''
         );
+        yarnPath = null;
+      }
+      if (yarnrc !== scrubbedYarnrc) {
+        logger.debug(`Writing scrubbed .yarnrc to ${lockFileDir}`);
         await writeLocalFile(
           upath.join(lockFileDir, '.yarnrc'),
           scrubbedYarnrc
         );
-        yarnPath = null;
```

Comment:
why this change?

Reply:
It is moved to above. I wanted to preserve its location at the end of the `if (!yarnBinaryExists)` block, because it felt related to that.

Happy to change it if you think that that makes more sense though.

Reply:
no, it's just sometimes hard to follow changes. leave as is